TestApps/SIL.Windows.Forms.TestApp/ContributorsForm.cs

```diff
@@ -0,0 +1,176 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Drawing;
+using System.Windows.Forms;
+using SIL.Windows.Forms.ClearShare;
+using SIL.Windows.Forms.ClearShare.WinFormsUI;
+using SIL.Windows.Forms.Widgets.BetterGrid;
+
+namespace SIL.Windows.Forms.TestApp
+{
+	public partial class ContributorsForm : Form
+	{
+		private TableLayoutPanel _tableLayout;
+		private ContributorsListControl _contributorsControl;
+		private ContributorsListControlViewModel _model;
+		private BetterGrid _contributorNames;
+
+		private class AutoCompleter : IAutoCompleteValueProvider
+		{
+			public BetterGrid Source {get; set;}
+
+			public IEnumerable<string> GetValuesForKey(string key)
+			{
+				if (key == "person")
+				{
+					foreach (DataGridViewRow row in Source.Rows)
+					{
+						yield return row.Cells[0].Value as string;
+					}
+				}
```

Comment:
Don't we have to return something if `key != "person"` ?

Reply:
Because it's an IEnumerable, not returning anything (or not hitting the yield return) is okay. That will just result in an empty enumeration. In this particular instance, if you look at the caller, the key will currently always be "person," but I put that check into the implementation because if it ever changes, it would presumably not make sense to return the same values.